sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/util/SearchPagedFlux.java

```diff
@@ -17,13 +20,16 @@
  * element type is {@link SearchResult}, and the page type is {@link SearchPagedResponse}.
  */
 public final class SearchPagedFlux extends PagedFluxBase<SearchResult, SearchPagedResponse> {
+    private final Supplier<Mono<SearchPagedResponse>> firstPageRetriever;
```

Comment:
I'm not certain this is the right pattern to be using. Looking at the method calls below to get the overall information will trigger this service call, and as @brjohnstmsft mentioned this information is variable between calls. I believe the pattern we should be following is something similar to what @anuchandy used when creating `PollerFlux`, in this case the first response will be cached in this class instead of the most recent response.

The logic would mutate slightly here, potentially making this class not thread-safe, to the following:

- Create a property on the class that will be set on first page retrieved.
- Wrap the passed supplier in another supplier that will set the variable.
- If one of the property methods on the class are called before the first page is retrieved it will get the first page and cache it, when the Flux is iterator we join that page with all the next page retrievals.

There are a few more issues here given how Java's paging mechanism works:

- What happens when the consumer begins their iteration using a sync token.
- How does this fit into the reactive stream concept.

The former may be resolved by having this class override those methods to better handle it, but there still is an issue that this Flux could be used multiple times with different starting locations.

```java
SearchPagedFlux searchFlux = search("search string);

searchFlux.subscribe(/* iterate by item */);

/* Some code later */

searchFlux.byPage(continuationToken).subscribe(/* iterate pages from starting token */);
```

In this case the `SearchPagedFlux` has been retained and will contain top level data for the first invocation, but there is not reason it couldn't be reused later.

As seen above there is also a pattern mismatch with how these top level properties need to be accessed. The consumer either needs to explicitly retain access to the `PagedFlux` or they need to use the directly, both are a bit of an anti-pattern for reactive programming.

Some of the issues raised here are more corner cases but I want to make sure we think through all the corner scenarios this concept adds and how it interacts with Java's usage of Reactor.

Reply:
Based on the offline discussion, we are going to pause the PR with the answer of
`Whether customer is using metadata (total count) together with paging result or use it independently?`

**Option 1:** (Use it together)
We will have such paging response:
```
SearchPagedResponse
{
      private Long totalCount; // with public getter, setter
      private Double coverage; // with public getter, setter
      private Map facetResult; // with public getter, setter
      private SearchResultItemContext(result item, metadata...) // with public getter, setter
}
```

**Option 2:** (Independent use)
We will have such paging flux:
```
SearchPagedFlux 
{
        private Supplier firstPageRetriver;
        private Long firstPagedCount;

        public synchronized Mono<Double> getCoverage() {
            if (firstPageCoverage != null) {
                return Mono.just(firstPageCoverage);
            }

            return firstPageRetriever.get().flatMap(response -> {
                if (response.getCoverage() == null) {
                    return Mono.empty();
                }
                firstPageCoverage = response.getCoverage();
                return Mono.just(firstPageCoverage);
             });
        }
}
```

Reply:
@brjohnstmsft